### PR TITLE
Some fixes, Tweaks and Charpter 13 support

### PR DESCRIPTION
**Pull request by JustShinobi**
- Refactored and simplified a few lines of code.
- Fix stuck problem when monthly login message appears, 
- Add charpter 13 support
- Swipe optimization of some maps like 11-3, 12-2, 12-3, 13-1, 13-2, 13-3, 13-4. In my tests, I was able to farm them smoothly, however it needs a lot of testing to be sure because of the RNG.
- New button_confirm asset.

### CODE REVIEW

**Comment by Al12rs** [comment]
Hey, thanks for your PR. I have some questions and feedback regarding your changes though. 

First it would be better if this kind of PR didn't contain formatting changes as those are subjective and a project could have specific policies about them. It makes it more difficult to identify the actual changes that were made to achieve the implemented feature. They basically add clutter and are unrelated so they should be submitted as a separate PR if needed.

Now regarding your actual changes I will make some comments directly under the code

File: ALAuto.py [hunk 1]
```diff
@@ -200,11 +202,31 @@ def print_cycle_stats(self):
...
+            Utils.script_sleep(1)
+            continue
+        if Utils.find("menu/announcement"):
+            Logger.log_msg("Found Announcement Window")
+            Utils.find_and_touch("menu/announcement")
+            Utils.script_sleep(1)
+            continue
+        if Utils.find("menu/alert_info"):
+            Logger.log_msg("Found alert.")
+            Utils.find_and_touch("menu/alert_close")
```
Comment (Al12rs) [inline ALAuto.py]: Why is this necessary, does not the current implementation already go back to the main menu?

File: modules/combat.py [hunk 2]
```diff
@@ -385,12 +390,27 @@ def clear_map(self):
...
+            elif cmap == '12-3':
+                Utils.swipe(1250, 530, 1300, 540, 100)
+            elif cmap == '13-1':
+                Utils.swipe(1020, 500, 1300, 540, 100)
+            elif cmap == '13-2':
+                Utils.swipe(1125, 550, 1300, 540, 100)
+            elif cmap == '13-3':
+                Utils.swipe(1150, 510, 1300, 540, 100)
+            elif cmap == '13-4':
+                Utils.swipe(1200, 460, 1300, 540, 100)
```
Comment (Al12rs) [inline modules/combat.py]: Wouldn't this be better to be implemented as a dictionary of swipe arguments?
Like so: `swipe(**args['cmap'])`

File: modules/commission.py [hunk 3]
```diff
@@ -46,12 +49,34 @@ def commission_logic_wrapper(self):
...
+                Utils.find_and_touch("menu/announcement")
+                Utils.script_sleep(1)
+                continue
+            if Utils.find("menu/alert_info"):
+                Logger.log_msg("Found alert.")
+                Utils.find_and_touch("menu/alert_close")
+                Utils.script_sleep(1)
+
+            if Utils.find("commission/button_completed") and (lambda x: 332 < x < 511)(
+                    Utils.find("commission/button_completed").y):
```
Comment (Al12rs) [inline modules/commission.py]: Why is this here?

File: ALAuto.py (hunk 1 above)
Comment (JustShinobi) [reply ALAuto.py]: The current implementation just clicks on a point in the upper left if it does not detect the button_battle asset , I think it quite incomplete and opens the window for several failures.

File: modules/combat.py (hunk 2 above)
Comment (JustShinobi) [reply modules/combat.py]: Actually I was thinking of implementing the same function used in the dorm, however I needed something functional urgent on these maps, it was the best solution I had and it didn't take me 10 seconds to think about it.

File: modules/commission.py (hunk 3 above)
Comment (JustShinobi) [reply modules/commission.py]: If you get idle just doing the commissions, when the server time resets and these messages pop up the bot get sutck, if youy game is open the message only appears when you start clicking in the game, so when you start commission module the bot get stuck.

**Comment by JustShinobi** [comment]
> Hey, thanks for your PR. I have some questions and feedback regarding your changes though.
> 
> First it would be better if this kind of PR didn't contain formatting changes as those are subjective and a project could have specific policies about them. It makes it more difficult to identify the actual changes that were made to achieve the implemented feature. They basically add clutter and are unrelated so they should be submitted as a separate PR if needed.
> 
> Now regarding your actual changes I will make some comments directly under the code

Sure, I understand your point, I've been studying python for quite some time now, and since I'm learning now I'm trying to follow the documentation faithfully.

Thanks for the feedback.

File: ALAuto.py (hunk 1 above)
Comment (JustShinobi) [reply ALAuto.py]: For example, if this popup sho up, the current implementation will not fix, already woke up a few times and saw the bot stuck in this message.

File: .gitignore [hunk 4]
```diff
@@ -109,4 +109,22 @@ venv.bak/
...
+.idea/vcs.xml
+
+
+config_dev.ini
+screenshot.png
+screenshot1.png
+screenshot2.png
+screenshot3.png
+screenshot4.png
+screenshot5.png
```
Comment (Egoistically) [inline .gitignore]: You could just add the whole .idea folder rather than every single file on it by adding `.idea/`. Also, why use config_dev.ini when there's already a config-local.ini added, just rename your existing one. 
Finally, I think all those screenshots could be moved into another folder, whether inside the project folder or not, to avoid adding each single one of them.

File: .idea/.gitignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+# Default ignored files
+/workspace.xml
```
Comment (Egoistically) [inline .idea/.gitignore]: What does this do?

File: ALAuto.py (hunk 1 above)
Comment (Egoistically) [inline ALAuto.py:227]: Please avoid formatting changes, I like how the current structure is and PRs that change it are instantly a big no no.

Regarding your addition, though: the code should be fine and it is not a bad implementation per se, but I'd like to test any changes to the main file myself. Also the fact that this notification could pop up while other modules are running makes this solution impractical.

File: ALAuto.py [hunk 6]
```diff
@@ -219,7 +241,7 @@ def print_cycle_stats(self):
             script.print_cycle_stats()
         else:
             Logger.log_msg("Nothing to do, will check again in a few minutes.")
-            Utils.script_sleep(300)
+            Utils.script_sleep(60)
```
Comment (Egoistically) [inline ALAuto.py]: Why?

File: modules/combat.py [hunk 7]
```diff
@@ -80,9 +84,9 @@ def combat_logic_wrapper(self):
                 Utils.touch_randomly(self.region["menu_button_battle"])
                 Utils.wait_update_screen(1)
                 continue
-            if Utils.find("combat/menu_fleet") and (lambda x:x > 414 and x < 584)(Utils.find("combat/menu_fleet").y):
+            if Utils.find("combat/menu_fleet") and (lambda x: 414 < x < 584)(Utils.find("combat/menu_fleet").y):
```
Comment (Egoistically) [inline modules/combat.py]: I don't like this change, it looks really obnoxious to me.

File: modules/combat.py (hunk 2 above)
Comment (Egoistically) [inline modules/combat.py]: As @Al12rs said, this could be better implemented by using a dictionary, I originally made it an `if` because it was only for a single map.

File: modules/combat.py [hunk 8]
```diff
@@ -471,20 +491,36 @@ def get_enemies(self, blacklist=[]):
...
+            l5 = filter(lambda x: x[1] > 80 and x[1] < 977 and x[0] > 180, map(lambda x: [x[0] + 75, x[1] + 110],
+                                                                               Utils.find_all('enemy/fleet_3_down',
+                                                                                              sim - 0.06,
+                                                                                              self.chapter_map)))
             l5 = [x for x in l5 if (not self.filter_blacklist(x, blacklist))]
-            l6 = filter(lambda x:x[1] > 80 and x[1] < 977 and x[0] > 180, map(lambda x:[x[0] + 75, x[1] + 110], Utils.find_all('enemy/fleet_2_up', sim - 0.06, self.chapter_map)))
+            l6 = filter(lambda x: x[1] > 80 and x[1] < 977 and x[0] > 180, map(lambda x: [x[0] + 75, x[1] + 110],
+                                                                               Utils.find_all('enemy/fleet_2_up',
+                                                                                              sim - 0.06,
+                                                                                              self.chapter_map)))
```
Comment (Egoistically) [inline modules/combat.py]: Again, please avoid formatting changes like these.

File: modules/combat.py [hunk 9]
```diff
@@ -1,11 +1,15 @@
...
+
+from scipy import spatial
 from util.logger import Logger
 from util.utils import Region, Utils
-from scipy import spatial
-from threading import Thread
+
 
 class CombatModule(object):
+    l: List[Any]
```
Comment (Egoistically) [inline modules/combat.py]: Subjective changes on the order and I don't comprehend why would you import `List` neither the last line is for. 
Sorry for my poor knowledge about the topic.

File: modules/commission.py (hunk 3 above)
Comment (Egoistically) [inline modules/commission.py]: What I told you on the other one, I think there are better ways to implement this to avoid repeating the same code on multiple files.

File: modules/commission.py [hunk 10]
```diff
@@ -81,7 +107,7 @@ def completed_handler(self):
                 self.stats.increment_commissions_received()
                 continue
             if Utils.find("menu/item_found"):
-                Utils.touch_randomly(self.region["tap_to_continue"])
+                Utils.find_and_touch("menu/tap_to_continue")
```
Comment (Egoistically) [inline modules/commission.py:107]: Why?

File: util/utils.py [hunk 11]
```diff
@@ -231,7 +235,7 @@ def find_all(cls, image, similarity=DEFAULT_SIMILARITY, cmap=None):
 
     @classmethod
     def match_resize(cls, image, scale, similarity=DEFAULT_SIMILARITY):
-        template_resize = cv2.resize(image, (), fx = scale, fy = scale, interpolation = cv2.INTER_NEAREST)
+        template_resize = cv2.resize(image, (), fx=scale, fy=scale, interpolation=cv2.INTER_NEAREST)
         match_resize = cv2.matchTemplate(screen, template_resize, cv2.TM_CCOEFF_NORMED)
         numpy.append(cls.locations, numpy.where(match_resize >= similarity))
```
Comment (Egoistically) [inline util/utils.py]: Every single change on this file is subjective.

**Review by Egoistically — CHANGES_REQUESTED** [review]
Please try to stack all changes on a single commit rather than pushing every single change.
I am really grateful for your work done, but I hope you understand I cannot merge this for a lot of reasons. Please fix what I proposed on my review and thank you very much for your work.

**Comment by Egoistically** [comment]
First time I write a PR review. I apparently didn't quote some parts of the code properly, but I think you can understand what I mean. Let me know if you don't understand something.

File: .gitignore (hunk 4 above)
Comment (JustShinobi) [reply .gitignore]: i'll fix, sorry.

File: .idea/.gitignore (hunk 5 above)
Comment (JustShinobi) [reply .idea/.gitignore]: for nothing, my mistake.

File: ALAuto.py (hunk 6 above)
Comment (JustShinobi) [reply ALAuto.py]: use this for local test, I did not realize that I sent it to the master branch, but I also see no reason not to decrease the default time, 300 is a very high value no?

File: modules/combat.py (hunk 7 above)
Comment (JustShinobi) [reply modules/combat.py]: alright, will revert.

File: modules/commission.py (hunk 10 above)
Comment (JustShinobi) [reply modules/commission.py:107]: Because if we receive many items the position of the button will change, I have not had problems with it so far, because a click anywhere on the screen makes the popup close, but I think this implementation more reliable.

**Comment by JustShinobi** [comment]
> First time I write a PR review. I apparently didn't quote some parts of the code properly, but I think you can understand what I mean. Let me know if you don't understand something.

Thanks for feedback, i' will do some fixes.

File: ALAuto.py (hunk 6 above)
Comment (Egoistically) [reply ALAuto.py]: Ya, back when I set 5 minutes I didn't know that script_sleep randomized the time.

File: modules/commission.py (hunk 10 above)
Comment (Egoistically) [reply modules/commission.py:107]: Ohhhh, that makes sense. I will check it out later, thanks!

File: ALAuto.py (hunk 1 above)
Comment (JustShinobi) [reply ALAuto.py:227]: The purpose of my implementation is just to keep the bot running on the commission module, I see no reason to work with other modules.
